### PR TITLE
[M2P-486] Add feature switch for Bolt SSO and a skeleton of the new endpoint that just returns a 500

### PR DESCRIPTION
**Pull request by adentong**
# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: (link Jira ticket)

#changelog [MA-528] Add feature switch for Bolt SSO and a skeleton of the new endpoint that just returns a 500

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [x] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [x] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

[MA-528]: https://boltpay.atlassian.net/browse/MA-528

### CODE REVIEW

File: Model/Api/OAuthRedirect.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     * @api
+     *
+     * @param string $authorization_code
+     *
+     * @return void
+     */
+    public function login($authorization_code = '')
+    {
+        if (!$this->deciderHelper->isBoltSSOEnabled()) {
+            $this->sendErrorResponse('Bolt SSO feature not enabled');
```
Comment (nickfiacco-bolt) [inline Model/Api/OAuthRedirect.php:60]: I don't think we want to say this in the message. What does it say if you make a request to an endpoint that doesn't exist?

**Review by nickfiacco-bolt — COMMENTED** [review]
Mostly LGTM, but update the error message to not say anything (we don't want to reveal internals). Also so many formatting changes 😱

**Comment by adentong** [comment]
> Mostly LGTM, but update the error message to not say anything (we don't want to reveal internals). Also so many formatting changes 😱

Lol yea that's my php formatter. I'll update the error message.

File: etc/di.xml [hunk 2]
```diff
@@ -69,6 +69,9 @@
     <!-- For debug endpoint -->
     <preference for="Bolt\Boltpay\Api\DebugInterface" type="Bolt\Boltpay\Model\Api\Debug" />
 
+    <!-- For debug endpoint -->
```
Comment (vitaliyreznikov) [inline etc/di.xml:72]: please fix the comment

**Review by vitaliyreznikov — APPROVED** [review]

**Comment by danac-gs** [comment]
@adentong @vitaliyreznikov May get the access to the task in Jira?

**Review by danac-gs — APPROVED** [review]
LGTM

File: etc/di.xml (hunk 2 above)
Comment (adentong) [reply etc/di.xml:72]: Oops bad copy and paste

**Comment by adentong** [comment]
@danac-gs i will move the task to m2 project so you have access

**Review by vitaliyreznikov — APPROVED** [review]

**Pull request merged**